test/integration_download_test.go

```diff
@@ -144,6 +145,17 @@ func TestRemoteDownload(t *testing.T) {
 	runTerragrunt(t, fmt.Sprintf("terragrunt apply -auto-approve --terragrunt-non-interactive --terragrunt-working-dir %s", testFixtureRemoteDownloadPath))
 }
 
+func TestInvalidRemoteDownload(t *testing.T) {
```

Comment:
```
time=2023-12-15T16:57:27Z level=debug msg=Enabling enforced TLS access for S3 bucket terragrunt-test-bucket-0n6bas prefix=[/tmp/terragrunt-parent-child-test516286081/stage/my-app] 
=== NAME  TestInvalidRemoteDownload
    integration_test.go:4097: Failed to run Terragrunt command 'terragrunt apply -auto-approve --terragrunt-non-interactive --terragrunt-working-dir fixture-download/remote-invalid' due to error: downloading source url https://github.com/totallyfakedoesnotexist/notreal.git?ref=v1.2.3
        1 error occurred:
        	* bad response code: 404
        
        
        
        Stdout: (see log output above)
        
        Stderr: (see log output above)
--- FAIL: TestInvalidRemoteDownload (2.28s)

```

Reply:
@denis256 thanks for taking the time to review. I updated the test section as suggested. please review. many thanks